packages/sdk/src/Actions/EventAction.ts

```diff
@@ -526,21 +529,85 @@ export class EventAction extends DeployableTarget<
    * @public
    * @async
    * @param {ActionStep} actionStep - The action step to validate. Can be a function of event step.
-   * @param {?TxParams & { chainId?: number }} [params] - Additional parameters for validation, including known events, logs, and chain ID.
+   * @param {TxParams & { transactionHash?: Hex, chainId?: number }} params - Additional parameters for validation, including, hash, known events, logs, and chain ID.
    * @returns {Promise<boolean>}
    */
   public async isActionStepValid(
     actionStep: ActionStep,
-    params?: TxParams & { chainId?: number },
+    params: TxParams & { hash?: Hex; chainId?: number },
   ) {
+    const client = this._config.getClient({
+      chainId: params.chainId,
+    }) as PublicClient;
+
     if (actionStep.signatureType === SignatureType.EVENT) {
-      return await this.isActionEventValid(actionStep, params);
+      const eventParams = params as ValidateEventStepParams;
+      const signature = actionStep.signature;
+      let event: AbiEvent;
+      // Lookup ABI based on event signature
+      if (eventParams.knownEvents) {
+        event = eventParams.knownEvents?.[signature] as AbiEvent;
+      } else {
+        event = (events.abi as Record<Hex, AbiEvent>)[signature] as AbiEvent;
+      }
+
+      if (!event) {
+        throw new Error(`No known ABI for given event signature: ${signature}`);
+      }
+
+      if (this.isArraylikeIndexed(actionStep, event)) {
+        // If the field is indexed, we can't filter on it
+        throw new UnparseableAbiParamError(
+          actionStep.actionParameter.fieldIndex,
+          event,
+        );
+      }
+
+      eventParams.event = event;
+
+      // Use the provided logs, no need to fetch receipt
+      if (eventParams.logs) {
+        return this.isActionEventValid(actionStep, {
+          ...eventParams,
+        });
+      }
+
+      if (!params?.hash) {
+        throw new Error('Hash is required for event validation');
```

Comment:
let's make all of these typed errors

Reply:
something prefixed with `Validation` maybe, `new ValidationHashMissingError`, `ValidationChainIdMissingError`?